include/exadg/grid/grid_utilities.h

```diff
@@ -245,6 +246,52 @@ create_fine_and_coarse_triangulations(
   }
 }
 
+/**
+ * This function reads an external triangulation. The path to the triangulation file should
+ * be defined in the input file and then stored in GridData. This function extracts it from
+ * GridData and stores it in the triangulation "tria".
+ */
+template<int dim>
+inline void
+read_external_triangulation(dealii::Triangulation<dim, dim> & tria, GridData const & data)
+{
+  AssertThrow(!data.file_name.empty(),
+              dealii::ExcMessage(
+                "You are trying to read a grid file, but the string, which is supposed to contain"
+                " the file name, is empty. Most likely, you forgot to specify the file location in"
+                " the input file."));
+
+  dealii::GridIn<dim> grid_in;
+
+  grid_in.attach_triangulation(tria);
+
+  // find the file extension from the file name
+  std::string extension = data.file_name.substr(data.file_name.find_last_of('.') + 1);
+
+  AssertThrow(!extension.empty(),
+              dealii::ExcMessage("You are trying to read a grid file, but the file extension is"
+                                 " empty. Check the input file to make sure that the file"
+                                 " extension is correctly defined after a full stop."));
+
+  // decide the file format
+  typename dealii::GridIn<dim>::Format format;
+  if(extension == "e" || extension == "exo")
+    format = dealii::GridIn<dim>::Format::exodusii;
+  else
+    format = grid_in.parse_format(extension);
+
+  // TODO: check if the exodusIIData is needed
+  // typename dealii::GridIn<dim>::ExodusIIData exodusIIData;
+
+  grid_in.read(data.file_name, format);
+
+  AssertThrow(get_element_type(tria) == data.element_type,
+              dealii::ExcMessage("You are trying to read a grid file, but the element type of the"
+                                 " external grid file and the element type specified in GridData"
+                                 " don't match. Most likely, you forgot to change the element_type"
+                                 " parameter of GridData to the desired element type."));
```

Comment:
I also do not see why this assert has to be placed in `read_external_triangulation()`. As the function name says, a grid file should be read (think about single-responsibility principle: who is responsible for reading the grid? who is responsible for checking consistency of `GridData` content?). It is in my opinion not clear that it is the responsibility of this function to check compatibility/consistency with other data structures of ExaDG (such as `GridData`, where it is questionable if this struct should be passed at all to this function). I understand the goal to write this code only once in the code, but the current solution appears to be sub-optimal in my opinion.

Reply:
If you are undecided regarding this point, I would suggest to keep it as is for now and potentially change it later.

Reply:
I see, what you mean. I think my comment above https://github.com/exadg/exadg/pull/323#discussion_r1105533234 explains why I decided to go for this solution. 


> I understand the goal to write this code only once in the code, but the current solution appears to be sub-optimal in my opinion.

Yes, this was my goal. And I can't think of another solution without duplicating code. Maybe something like `GridData.check()`, but I think this would complicate this PR.